Jenkinsfile

```diff
@@ -1,25 +1,26 @@
 // For ci.jenkins.io
 // https://github.com/jenkins-infra/documentation/blob/master/ci.adoc
 
-def mavenName = 'mvn'
-def jdkName = 'jdk8'
-
-node('docker') {
-    checkout scm
-
-    def uid = sh returnStdout: true, script: "id -u | tr -d '\n'"
-    def gid = sh returnStdout: true, script: "id -g | tr -d '\n'"
-
-    def buildArgs = "--build-arg=uid=${uid} --build-arg=gid=${gid} src/main/resources/ath-container"
-    def image = docker.build('jenkins/ath', buildArgs)
-
-    String containerArgs = '-v /var/run/docker.sock:/var/run/docker.sock'
-    image.inside(containerArgs) {
-        sh '''
-            eval $(./vnc.sh)
-            ./run.sh firefox latest -Dmaven.test.failure.ignore=true -DforkCount=1 -B
-        '''
+pipeline {
+    agent {
+        dockerfile {
+            dir 'src/main/resources/ath-container'
+            args '-v /var/run/docker.sock:/var/run/docker.sock'
```

Comment:
Does this ensure uid and gid will match that of the enclosing user?

Reply:
yep :smile: 


from the log:

```
$ docker run -t -d -u 1000:1000 -v /var/run/docker.sock:/var/run/docker.sock -w /home/jenkins/workspace/ce-test-harness_declarative-DDP25ZTXYUVWCZRCBHA2GTP6UAZH23SZWKFC236DSXZG6SGJE6JQ -v /home/jenkins/workspace/ce-test-harness_declarative-DDP25ZTXYUVWCZRCBHA2GTP6UAZH23SZWKFC236DSXZG6SGJE6JQ:/home/jenkins/workspace/ce-test-harness_declarative-DDP25ZTXYUVWCZRCBHA2GTP6UAZH23SZWKFC236DSXZG6SGJE6JQ:rw,z -v /home/jenkins/workspace/ce-test-harness_declarative-DDP25ZTXYUVWCZRCBHA2GTP6UAZH23SZWKFC236DSXZG6SGJE6JQ@tmp:/home/jenkins/workspace/ce-test-harness_declarative-DDP25ZTXYUVWCZRCBHA2GTP6UAZH23SZWKFC236DSXZG6SGJE6JQ@tmp:rw,z -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** --entrypoint cat ed75bee47b87e2ed1c73e22b6e1bce25c3ad236b
```